server/chatroom.py

```diff
@@ -20,36 +23,188 @@ def get_msg_update(cur, chatroom_id, date_time):
+        #make sure the target user is verified
+        cur.execute("SELECT student_id FROM student WHERE student_id = %(target_user_id)s",
+                    {"target_user_id" : target_user_id})
+        res = cur.fetchall()[0][0]
+        if (res != target_user_id):
+            raise HTTPException(status_code=500, detail="Student not verified!")
+        
+        #get target user of invite, check to see if it matches target_user_id parameter
+        cur.execute("SELECT target_user_id FROM invite WHERE invite_id=%(invite_id)s", {"invite_id" : invite_id})
+        res = cur.fetchall()[0][0]
+        if (res != target_user_id):
+            raise HTTPException(status_code=500, detail="The invite is not for this user")
+
+        
+        #cur.execute("SELECT id FROM chatroom WHERE invite_id = %(invite_id)s", {"invite_id" : invite_id})
+        cur.execute("SELECT chatroom_id FROM invite WHERE invite_id=%(invite_id)s", {"invite_id" : invite_id})
+        chatroom_id = cur.fetchall()[0][0]
+        
+        #do not allow insertion of user into in_chatroom if uni_id for target user and chatroom do not match
+        cur.execute("SELECT uni_id FROM chatroom WHERE id = %(chatroom_id)s", {"chatroom_id" : chatroom_id})
+        uni_id = cur.fetchall()[0][0]
+        cur.execute("SELECT uni_id FROM student WHERE student_id = %(target_user_id)s", {"target_user_id" : target_user_id})
+        
+        student_uni_id = cur.fetchall()[0][0]
+        
+        if (student_uni_id != uni_id):
+            raise HTTPException(status_code=500, detail="The university of the invitee does not match the university of the chatroom")
+        
+        #insert target user into in_chatroom
+        cur.execute("INSERT INTO in_chatroom(student_id, uni_id, chatroom_id) VALUES(%(target_user_id)s, %(uni_id)s, %(chatroom_id)s)",
+                    {"target_user_id" : target_user_id, "uni_id" : uni_id, "chatroom_id" : chatroom_id})
+        
+        return True
+    
+    except BaseException as e:
+        print(f'Exception: {e}')
+        raise HTTPException(
+            status_code=500,
+            detail="Database Error",
+        )
+
+# Get the list of chatrooms that a student is in (for visibility of chatrooms functionality)
+def get_chatrooms_for_student(cur, student_id):
+    try:
+        """
+        cur.execute("SELECT chatroom_name FROM (in_chatroom JOIN chatroom ON chatroom_id = id) WHERE student_id = %(student_id)s",
+                     {"student_id" : student_id})
+        """
+        cur.execute("SELECT (chatroom_name, chatroom.id) FROM (in_chatroom JOIN chatroom ON chatroom_id = id) WHERE student_id = %(student_id)s",
+                     {"student_id" : student_id})
+        #cur.execute("SELECT chatroom_name FROM chatroom")
+        res = cur.fetchall()
+        chatroom_names = [x[0][0] for x in res]
+        chatroom_ids = [x[0][1] for x in res]
+        chatrooms = {"chatroom_names" : chatroom_names, "chatroom_ids" : chatroom_ids}
```

Comment:
u dont hv to do this, its easier for js to leave it as a tuple. but i think u can scrap this function all together cos i dont think get_chatroom need any changes